recipes/trexio/meta.yaml

```diff
@@ -0,0 +1,58 @@
+
+{% set name = "trexio" %}
+{% set version = "2.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/TREX-CoE/trexio/releases/download/v{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 232866c943b98fa8a42d34b55e940f7501634eb5bd426555ba970f5c09775e83
+
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - autoconf  
+    - automake  
+    - libtool   
+    - make      
+    - gnuconfig 
+    - pkg-config
+
+  host:
+    - hdf5
+    # the dependencies below are not required by TREXIO but they get exported from HDF5
+    # and if they are not listed here than the `conda-build --error-overlinking` fails
+    - zlib    
+    - openssl 
+    - libcurl 
```

Comment:
```
  ERROR (trexio,lib/libtrexio.so.0.0.0): Needed DSO lib/libcurl.so.4 found in ['libcurl']
  ERROR (trexio,lib/libtrexio.so.0.0.0): .. but ['libcurl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)
```
This error message from conda means that it has inspected this binary and found that it links to libcurl (like when you inspect a file with ldd). Since, you state in this comment that libcurl is not a dependency of libtrexio, you should double check your build process to prevent overlinking.

You should *not* add the packages to the host section if trexio doesn't need them. If they are needed by hdf5 at runtime, hdf5 will have a run requirement for them.

Reply:
Indeed, hdf5 has a run requirement for these packages (see their [meta.yaml](https://github.com/conda-forge/hdf5-feedstock/blob/master/recipe/meta.yaml)). Our build system based on Autotools parses the libraries that are linked against HDF5 and adds them to the `LIBS` list, which results in overlinking. I switched to the CMake build and there is no overlinking issue. Probably the `FindHDF5.cmake` module was designed with these things in mind.